catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -59,55 +60,72 @@
 
     private int subDirLevels;
 
+    private PeriodicBatchExecutor<Metacard> periodicDrain;
+
     private boolean enableBackupPlugin = true;
 
     public CatalogBackupPlugin() {
+
+        init();
+    }
+
+    private void init() {
         subDirLevels = 0;
+        setPeriodicDrain(new PeriodicBatchExecutor<>(1000, 60, TimeUnit.SECONDS));
+        getPeriodicDrain().execute(this::backup);
     }
 
     /**
      * Backs up created metacards to the file system backup.
      *
      * @param input the {@link CreateResponse} to process
-     * @return
+     * @return {@link CreateResponse}
      * @throws PluginExecutionException
      */
     @Override
-    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+    public synchronized CreateResponse process(CreateResponse input)
```

Comment:
If we thread out metacard creation, don't we need to also thread out the update and deletes and make sure things are kept in the right order? Otherwise, we could end up trying to delete and/or update a metacard that's not been created yet.